src/crimson/os/cyanstore/cyan_store.cc

```diff
@@ -703,6 +711,30 @@ int CyanStore::_truncate(const coll_t& cid, const ghobject_t& oid, uint64_t size
   return r;
 }
 
+int CyanStore::_clone(const coll_t& cid, const ghobject_t& oid,
+                      const ghobject_t& noid)
+{
+  logger().debug("{} cid={} oid={} noid={}",
+                __func__, cid, oid, noid);
+  auto c = _get_collection(cid);
+  if (!c)
+    return -ENOENT;
+
+  ObjectRef oo = c->get_object(oid);
+  if (!oo)
+    return -ENOENT;
+  if (local_conf()->memstore_debug_omit_block_device_write)
+    return 0;
+  ObjectRef no = c->get_or_create_object(noid);
+  used_bytes += oo->get_size() - no->get_size();
```

Comment:
This won't behave correctly if oo->get_size() < no->get_size().

Reply:
Should we add `oo->get_size()` only? Since we discard the previous content (in case the object already existed).

Reply:
No, I mean that get_size() returns size_t, an unsigned integer which will underflow to a very large positive integer if oo->get_size() < no->get_size().  You need to cast each to a signed type (ssize_t) first.

Reply:
Fixed.